v2/clients/http/utils/common.go

```diff
@@ -79,18 +81,32 @@ func createRequest(ctx context.Context, httpMethod string, baseUrl string, reque
 	return req, nil
 }
 
-func createRequestWithRawData(ctx context.Context, httpMethod string, url string, data interface{}) (*http.Request, errors.EdgeX) {
-	jsonEncodedData, err := json.Marshal(data)
-	if err != nil {
-		return nil, errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to encode input data to JSON", err)
+func createRequestWithRawData(ctx context.Context, httpMethod string, url string, data interface{}, cborEncoding bool) (*http.Request, errors.EdgeX) {
```

Comment:
don't use bool because we might have other encoding in the future.
please use string and leverage clients.ContentTypeCBOR / clients.ContentTypeJSON.